integration-tests/src/test/java/com/datastax/oss/driver/mapper/NamingStrategyIT.java

```diff
@@ -0,0 +1,269 @@
+ *
+ * <ul>
+ *   <li>{@link DefaultStrategyEntity default}
+ *   <li>{@link UpperSnakeCaseEntity non-default built-in convention}
+ *   <li>{@link NameConverterEntity custom name converter class}
+ *   <li>{@link CustomNamesEntity custom names provided through annotations}
+ * </ul>
+ *
+ * <p>See each entity's corresponding table schema in {@link #setup()}.
+ */
+@Category(ParallelizableTests.class)
+public class NamingStrategyIT {
+
+  private static CcmRule ccm = CcmRule.getInstance();
+
+  private static SessionRule<CqlSession> sessionRule = SessionRule.builder(ccm).build();
+
+  @ClassRule public static TestRule chain = RuleChain.outerRule(ccm).around(sessionRule);
+
+  private static TestMapper mapper;
+
+  @BeforeClass
+  public static void setup() {
+    CqlSession session = sessionRule.session();
+
+    for (String query :
+        ImmutableList.of(
+            "CREATE TABLE default_strategy_entity(entity_id int primary key)",
+            "CREATE TABLE \"UPPER_SNAKE_CASE_ENTITY\"(\"ENTITY_ID\" int primary key)",
+            "CREATE TABLE test_NameConverterEntity(test_entityId int primary key)",
+            "CREATE TABLE custom_entity(custom_id int primary key)")) {
+      session.execute(
+          SimpleStatement.builder(query).setExecutionProfile(sessionRule.slowProfile()).build());
+    }
+
+    mapper = new NamingStrategyIT_TestMapperBuilder(session).build();
+  }
+
+  @Test
+  public void should_map_entity_with_default_naming_strategy() {
+    DefaultStrategyEntityDao dao = mapper.defaultStrategyEntityDao(sessionRule.keyspace());
+    DefaultStrategyEntity entity = new DefaultStrategyEntity(1);
+
+    dao.save(entity);
+    DefaultStrategyEntity retrievedEntity = dao.findById(1);
+    assertThat(retrievedEntity.getEntityId()).isEqualTo(1);
+  }
+
+  @Test
+  public void should_map_entity_with_non_default_convention() {
+    UpperSnakeCaseEntityDao dao = mapper.upperSnakeCaseEntityDao(sessionRule.keyspace());
+    UpperSnakeCaseEntity entity = new UpperSnakeCaseEntity(1);
+
+    dao.save(entity);
+    UpperSnakeCaseEntity retrievedEntity = dao.findById(1);
+    assertThat(retrievedEntity.getEntityId()).isEqualTo(1);
+  }
+
+  @Test
+  public void should_map_entity_with_name_converter() {
+    NameConverterEntityDao dao = mapper.nameConverterEntityDao(sessionRule.keyspace());
+    NameConverterEntity entity = new NameConverterEntity(1);
+
+    dao.save(entity);
+    NameConverterEntity retrievedEntity = dao.findById(1);
+    assertThat(retrievedEntity.getEntityId()).isEqualTo(1);
+  }
+
+  @Test
+  public void should_map_entity_with_custom_names() {
+    CustomNamesEntityDao dao = mapper.customNamesEntityDao(sessionRule.keyspace());
+    CustomNamesEntity entity = new CustomNamesEntity(1);
+
+    dao.save(entity);
+    CustomNamesEntity retrievedEntity = dao.findById(1);
+    assertThat(retrievedEntity.getEntityId()).isEqualTo(1);
+  }
+
+  @Entity
+  public static class DefaultStrategyEntity {
```

Comment:
nit: I think all public keywords can be removed

Reply:
Yes, it would work for these  tests because everything lives in the same package.

But in real life mapper interfaces/entities will most likely be public, so it doesn't hurt to be a bit more realistic here.